### ISSUE TITLE
Error messages for invalid column name are not informative

### ISSUE THREAD

**Issue by david-cortes** [pola-rs/polars#4083]
If I try to do some operation on a non-existant column name and this operation is followed by a groupby+agg, the error message will oftentimes be non-informative.

Example:
```python
import pandas as pd, polars as pl
df = pd.DataFrame({
    "colA" : ["a","b","c"],
    "colB" : [1,2,3]
})
(
    pl
    .DataFrame(df)
    .lazy()
    .filter(~pl.col("colC").is_null())
    .groupby(["colA"])
    .agg([
        pl.col("colB").sum().alias("result")
    ])
    .collect()
)
```
```
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
<ipython-input-1-151d8c8808b1> in <module>
      5 })
      6 (
----> 7     pl
      8     .DataFrame(df)
      9     .lazy()

~/anaconda3/lib/python3.9/site-packages/polars/internals/lazy_frame.py in collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, string_cache, no_optimization, slice_pushdown)
    630             slice_pushdown,
    631         )
--> 632         return self._dataframe_class._from_pydf(ldf.collect())
    633 
    634     def fetch(

PanicException: called `Option::unwrap()` on a `None` value
```

In other cases with longer chains, the error it ends up throwing says that the name of the aggregated column is "Not Found", while the issue is that it is either filtering or grouping by an invalid column.

Would be helpful if the error messages highlighted that a non-existant column is being referenced.

**Comment by ritchie46** [comment]
```
df = pd.DataFrame({
    "colA" : ["a","b","c"],
    "colB" : [1,2,3]
})
(
    pl
    .DataFrame(df)
    .lazy()
```

No need for a pandas indirection here. We support the same dataframe constructor here.

**Comment by ritchie46** [comment]
fixed by #4095
